src/board.js

```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react'
+import { Square } from './square.js'
+
+const board = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+export function Board(props) {
+  const [squareColors, setSquareColors] = useState(Array(board.length * board.length).fill('grey'))
+  const [clickedSquare, setClickedSquare] = useState(null)
+
+  const handleSquareClick = (i) => {
+    const squares_slice = squareColors.slice()
+    squares_slice[i] = 'red'
+    setSquareColors(squares_slice)
+    setClickedSquare(i)
+  }
+
+  const handleSquareHover = (i) => {
+    if (clickedSquare) {
+      const [hoveredRow, hoveredColumn] = squareIndexToRowColumn(i)
+      const [clickedRow, clickedColumn] = squareIndexToRowColumn(clickedSquare)
+
+      if (hoveredRow === clickedRow) {
+        const squares_slice = squareColors.slice()
+        squares_slice[i] = 'red'
+        setSquareColors(squares_slice)
+      }
+
+      if (hoveredColumn === clickedColumn) {
+        const squares_slice = squareColors.slice()
+        squares_slice[i] = 'red'
+        setSquareColors(squares_slice)
+      }
+    } 
+  }
+
+  const handleMouseUp = (_e) => setClickedSquare(null)
```

Comment:
Is there a way where I can cancel my selection? Cause this means as long as I am clicked I can just highlight everything in the row or column, right?

Reply:
Ya, I want to do that next, so if you drag back it unhighlights them